src/memory_manager.rs

```diff
@@ -350,6 +358,60 @@ pub fn get_allocator_mapping<VM: VMBinding>(
     mmtk.plan.get_allocator_mapping()[semantics]
 }
 
+/// Return an AllocatorInfo for the given allocator selector. This method is provided
+/// so that VM compilers may generate allocator fast-path and load fields for the fast-path.
+///
+/// Arguments:
+/// * `selector`: The allocator selector to query.
+pub fn get_allocator_info<VM: VMBinding>(selector: AllocatorSelector) -> AllocatorInfo {
+    match selector {
+        AllocatorSelector::BumpPointer(index) => {
+            let base_offset = offset_of!(Mutator<VM>, allocators)
+                + offset_of!(Allocators<VM>, bump_pointer)
+                + size_of::<BumpAllocator<VM>>() * index as usize;
+            let limit_offset = base_offset + offset_of!(BumpAllocator<VM>, limit);
+            let top_offset = base_offset + offset_of!(BumpAllocator<VM>, cursor);
+
+            AllocatorInfo::BumpPointer {
+                limit_offset,
+                top_offset,
+            }
+        }
+
+        AllocatorSelector::Immix(index) => {
+            let base_offset = offset_of!(Mutator<VM>, allocators)
+                + offset_of!(Allocators<VM>, immix)
+                + size_of::<ImmixAllocator<VM>>() * index as usize;
+            let limit_offset = base_offset + offset_of!(ImmixAllocator<VM>, limit);
+            let top_offset = base_offset + offset_of!(ImmixAllocator<VM>, cursor);
```

Comment:
I still think the name `cursor` is better than `top` because `top` may be interpreted as a synonym of limit.

Reply:
Updated